### PR TITLE
social media links title and description from socialdb app widget metadata

### LINKED ISSUE #135: Social media links should show title and description from socialdb app widget metadata
IMG_9541

_Originally posted by @frol in https://github.com/NEAR-DevHub/neardevhub-contract/pull/128#pullrequestreview-2127563410_

### PR DESCRIPTION
**Pull request by petersalomonsen**
Get title and description for social media  landing page from socialdb, by using web4 preloads querying social db.

You can verify the functionality in the deployment to `peterdevhub.near` where you can see that the description is set according to the data registered in socialdb.

```
curl https://peterdevhub.near.page
```

image

The preload url being used is:

https://rpc.web4.near.page/account/social.near/view/get?keys.json=[%22peterdevhub.near/widget/app/metadata/**%22]

where you can see that `name` and `description` from the returned payload matches the title and description in the preview environment metadata tags.

fixes #135

### CODE REVIEW

**Comment by elliotBraem** [comment]
Awesome development-- I saw this and I want to share some related details if they're helpful:

If "Index" is used instead of "app", then it will automatically get a url to HOT Telegram wallet. For example, pokethe.near/widget/Index has https://pokethe.bos.gg/ and https://t.me/herewalletbot/app?startapp=web-pokethe-bos-gg

I've been using "Index" from now on because of this. Although I wouldn't say it's a requirement at all, just a convenient feature.

bos-workspace also supports a "index" field in bos.config.json, see [here](https://github.com/NEARBuilders/bos-workspace?tab=readme-ov-file#configuration) -- it uses this index to initialize the src when passed a web-component dist, and can be used for [building and deploying to web4](https://github.com/NEARBuilders/bos-workspace/issues/104), from a bos-workspace. Could this json config also influence a web4contract deploy?

File: src/web4/handler.rs [hunk 1]
```diff
@@ -14,16 +14,59 @@ use crate::{
...
+    engine::GeneralPurpose::new(&alphabet::URL_SAFE, general_purpose::PAD);
 
 pub fn web4_get(contract: &Contract, request: Web4Request) -> Web4Response {
     let path_parts: Vec<&str> = request.path.split('/').collect();
 
     // A valid path provided by a legit web4 gateway always has '/', so there
     // are always [0] and [1] elements, and [0] is always empty.
     let page = path_parts[1];
+
+    let metadata_preload_url = format!("https://rpc.web4.near.page/account/social.near/view/get?keys.json=[%22{}/widget/app/metadata/**%22]", env::current_account_id());
```
Comment (frol) [inline src/web4/handler.rs]: You don't need to query specific RPC. Use `/web4/contract/...` API provided by web4 gateway:

image

If you have any further questions, ask them in https://t.me/web4near

**Review by frol — CHANGES_REQUESTED** [review]

File: src/web4/handler.rs (hunk 1 above)
Comment (petersalomonsen) [reply src/web4/handler.rs]: thanks! I should have seen this from the docs.

**Comment by frol** [comment]
@elliotBraem Hmm, the HOT link does not work for me. Does it work for you?

**Review by frol — APPROVED** [review]

**Comment by elliotBraem** [comment]
I missed this comment @frol -- you're right, it's not working for me anymore either. 

Asking in the telegram group: @bosintg

Edit:

> they've banned external links. Only a whitelist works

**Pull request merged**